packages/twenty-server/src/engine/core-modules/text-to-sql/text-to-sql.resolver.ts

```diff
@@ -0,0 +1,52 @@
+import { Args, Query, Resolver, ArgsType, Field } from '@nestjs/graphql';
+import { ForbiddenException, UseGuards } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { JwtAuthGuard } from 'src/engine/guards/jwt.auth.guard';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { AuthWorkspace } from 'src/engine/decorators/auth/auth-workspace.decorator';
+import { TextToSQLService } from 'src/engine/core-modules/text-to-sql/text-to-sql.service';
+import { TextToSQLQueryResult } from 'src/engine/core-modules/text-to-sql/dtos/text-to-sql-query-result.dto';
+import {
+  FeatureFlagEntity,
+  FeatureFlagKeys,
+} from 'src/engine/core-modules/feature-flag/feature-flag.entity';
+
+@ArgsType()
+class GetTextToSQLArgs {
+  @Field(() => String)
+  text: string;
+}
+
+@UseGuards(JwtAuthGuard)
+@Resolver(() => TextToSQLQueryResult)
+export class TextToSQLResolver {
```

Comment:
Might be slightly misleading as I initially though this would convert text to SQL while SQL is just the intermediate step, as I understand it this is a full fledged endpoint that takes the text and also return the query results